src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -168,10 +193,11 @@ public override Task<TextEdit[]> FormatSnippetAsync(Uri uri, DocumentSnapshot do
             }
 
             var originalText = context.SourceText;
-            var edits = result.Edits;
+            var edits = GetMinimalEdits(originalText, result.Edits);
```

Comment:
@davidwengier see description for more detail here but will this be troublesome?

Reply:
I'm surprised this makes any difference.. If we're not collapsing edits, then the code is essentially unchanged, and if we're collapsing edits, then it shouldn't matter either way since they'll all be merged together.

Am I missing something?

Reply:
Ah ya it made a huge difference because upon collapsing the edits it's trying to turn N number of edits into 1; however, if those edits aren't minimal the single edit that gets collapsed contains too much whitespace. In the case of completion where trailing whitespace can impact the cursor position the minimal edit call removes the "extra whitespace" if it already exists.

Reply:
oh yeah, that makes sense actually. I guess in my mind I was envisiaging the merge algorithm doing something to actually find the changes that the edits made, but if it operates purely on the edits themselves, I can see why this would help.